src/init.cpp

```diff
@@ -1581,8 +1582,11 @@ bool AppInitMain(boost::thread_group& threadGroup, CScheduler& scheduler)
 
     // ********************************************************* Step 8: load wallet
 #ifdef ENABLE_WALLET
-    if (!OpenWallets())
-        return false;
+    {
+        LOCK(cs_main);
+        if (!OpenWallets())
```

Comment:
I'd kinda prefer we push this lock down into wallet/init. Seemd weird to export the cs_main requirement to OpenWallets.